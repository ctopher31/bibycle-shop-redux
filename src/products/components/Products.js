import React, { Component } from 'react';
import ProductListing from './ProductListing';

class Products extends Component {
  componentDidMount() {
    this.props.loadProducts();
  }

  render() {
    const { items, addItem } = this.props;

    return (
      <main className="content shop">
        <h1 className="page--heading">Shop our Products</h1>
        <section className="product--section">
          {items.length < 1 ? (
            <div>Loading...</div>
          ) : (
            items.map(item => <ProductListing key={item.number} {...{ ...item, addItem }} />)
          )}
        </section>
      </main>
    );
  }
}

export default Products;
