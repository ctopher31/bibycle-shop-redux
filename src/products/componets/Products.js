import React from 'react';
import ProductListing from './ProductListing';

const Products = ({ products, addItem }) => (
  <main className="content shop">
    <h1 className="page--heading">Shop our Products</h1>
    <section className="product--section">
      {products.length < 1 ? (
        <div>Loading...</div>
      ) : (
        products.map(item => <ProductListing key={item.number} {...{ ...item, addItem }} />)
      )}
    </section>
  </main>
);

export default Products;
