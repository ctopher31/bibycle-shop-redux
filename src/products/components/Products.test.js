import React from 'react';
import { mount } from 'enzyme';
import Products from './Products';

describe('Products component', () => {
  it('renders correctly', () => {
    const products = [
      {
        number: '51001',
        name: 'widget',
        price: 35.57,
        salePrice: 32.95,
        onSale: true,
      },
    ];
    const addItem = () => 1;
    const wrapper = mount(<Products {...{ products, addItem }} />);

    expect(wrapper.exists('.content')).toBe(true);
    expect(wrapper.contains(<h1 className="page--heading">Shop our Products</h1>)).toBe(true);
    expect(wrapper.exists('.product--section')).toBe(true);
    expect(wrapper.props().products).toHaveLength(1);
    expect(wrapper.props().addItem()).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
