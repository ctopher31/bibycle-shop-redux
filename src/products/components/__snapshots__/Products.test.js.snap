// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component renders correctly 1`] = `ReactWrapper {}`;
